README.md

```diff
@@ -544,9 +625,25 @@ def count_transitions(data: HMMData) -> ArrayLike:
     k, _, z = data
     counts = np.zeros((k, k))  # How often each of the k*k state transitions
     # FIXME: count the transitions
-    for i in range(len(z) - 1):
-        counts[z[i], z[i+1]] += 1
+
+    ic = -1 #index_counter
+    #len(z)
+    match k:
+        case 3:
+            for _ in z:
+                ic += 1
+                if ic > 0:
+                    counts[z[ic-1]][z[ic]] = counts[z[ic-1]][z[ic]] + 1
+        case 7:
+            for _ in z:
+                ic += 1
+                if ic > 0:
+                    counts[z[ic-1]][z[ic]] = counts[z[ic-1]][z[ic]] + 1
```

Comment:
Aren't these two cases identical? In that case, there probably shouldn't be a `match` statement.